gravitee-kubernetes-client/src/main/java/io/gravitee/kubernetes/client/KubernetesClientV1Impl.java

```diff
@@ -240,8 +240,13 @@ public KubernetesClientV1Impl(Vertx vertx, KubernetesConfig kubernetesConfig) {
         Watch watch = new Watch();
         watchMap.putIfAbsent(watch.uid, watch);
 
+        String flowableKey = fieldSelector.equals("") ? resource.namespace : resource.name;
+        if (flowableMap.containsKey(flowableKey)) {
+            return (Flowable<T>) flowableMap.get(flowableKey);
```

Comment:
I think you should not reuse the flowable like this but, instead, it will be better to store the emitters instead and then you should be able to dispatch the events.

be sure also to cleanup the emitters when subscription is disposed.

Reply:
I'm not sure if I understood your comment properly. Here the idea is to re-use existing connection and have only one web-socket connection per namespace or resource and a ConnectableObservable is what we want.

Reply:
My bad, I didn't see you used the publish() and refCount() operators 👍